core/app/services/spree/import_service/products/update.rb

```
@@ -0,0 +1,62 @@
+
+        def refresh_properties
+          product.product_properties.delete_all
+          Spree::ImportService::Products::Serializers::PropertiesAttributesSerializer.new(row: row, product_id: product_id).to_a.tap do |attr|
+            Spree::ProductProperty.insert_all(attr)
+          end
+        end
+
+        def refresh_option_types
+          variant.option_value_variants.delete_all
+          product.product_option_types.delete_all
+          
```

RuboCop (1.75.5)

[convention] Trailing whitespace detected.

(Layout/TrailingWhitespace)